avogadro/io/mdlformat.cpp

```diff
@@ -210,13 +248,20 @@ bool MdlFormat::write(std::ostream& out, const Core::Molecule& mol)
   out << setw(3) << std::right << mol.atomCount() << setw(3) << mol.bondCount()
       << "  0  0  0  0  0  0  0  0999 V2000\n";
   // Atom block.
+  std::vector<std::pair<size_t, signed int>> chargeList;
   for (size_t i = 0; i < mol.atomCount(); ++i) {
     Atom atom = mol.atom(i);
+    signed int charge = atom.formalCharge();
+    chargeList.push_back(std::pair(atom.index(), charge));
+    unsigned int chargeField = (charge < 0)?
+        ((charge >= -3)? 4 - charge : 0) :
+        ((charge <= 3)? charge : 0);
     out << setw(10) << std::right << std::fixed << setprecision(4)
         << atom.position3d().x() << setw(10) << atom.position3d().y()
         << setw(10) << atom.position3d().z() << " " << setw(3) << std::left
-        << Elements::symbol(atom.atomicNumber())
-        << "  0  0  0  0  0  0  0  0  0  0  0  0\n";
+        << Elements::symbol(atom.atomicNumber()) << "0 "
```

Comment:
This isn't aligned correctly anymore. The symbol is fine, but the next zero should be " 0" (i.e., 2-character right-aligned) for the mass difference field.